# Predicting masked word using BETO 
In this section, learn how to use the pre-trained BETO model and predict the masked words in the Spanish text. 

Let's use the transformers pipeline API: 


In [1]:
%%capture
!pip install transformers==3.5.1

In [2]:
from transformers import pipeline


First, we define our masked word prediction pipeline. We pass the task which we want to perform, the pre-trained model, and tokenizer as arguments. As shown in the below code, we are using the dccuchile/bert-base-spanish-wwm-uncased model which is the pre-trained BETO model:


In [3]:
predict_mask = pipeline(
    "fill-mask",
    model= "dccuchile/bert-base-spanish-wwm-uncased",
    tokenizer="dccuchile/bert-base-spanish-wwm-uncased"
)


Let's take a sentence "todos los caminos llevan a roma" and mask the first word in the sentence with [MASK] token and feed the sentence with the masked token to our predict_mask pipeline and get the result: 


In [4]:
result = predict_mask('[MASK] los caminos llevan a Roma')


Print the result:

In [5]:
print(result)

[{'sequence': '[CLS] todos los caminos llevan a roma [SEP]', 'score': 0.9719983339309692, 'token': 1399, 'token_str': 'todos'}, {'sequence': '[CLS] todas los caminos llevan a roma [SEP]', 'score': 0.007171072065830231, 'token': 1825, 'token_str': 'todas'}, {'sequence': '[CLS] - los caminos llevan a roma [SEP]', 'score': 0.005352037958800793, 'token': 1139, 'token_str': '-'}, {'sequence': '[CLS] todo los caminos llevan a roma [SEP]', 'score': 0.004154080059379339, 'token': 1300, 'token_str': 'todo'}, {'sequence': '[CLS] y los caminos llevan a roma [SEP]', 'score': 0.00396431190893054, 'token': 1040, 'token_str': 'y'}]



From the above results, we can observe that we have scores, filled sequence, and also the predicted masked token (word). We can observe that our model has correctly predicted the masked word as 'todos' with a score of 0.97. 

Now that we learned how to use the BETO for Spanish, in the next section, let's look into BERTje. 